Fix note list toggle

codebykat opened the pull request:
### Fix
When we refactored the previousIndex (#1989) and note list, we didn't account for behavior on small screens. This PR disentangles that by clarifying when we want `CLOSE_NOTE` (unselect the selected note) and when we want `TOGGLE_NOTE_LIST` (keep a note selected, but switch to list view).

This has the interesting side effect of making it possible to maintain a selected note on the small screen. I removed the check that disables the selected note highlight on small views. You can now use the note list keyboard command to toggle between list and note view without losing the selected note, and use next/previous note from either view to select and open a new note.

This will fix #2029 

### Test
1. Load the app on a narrow window and ensure the note list shows up (not a blank screen)
2. Select a note. The note should show
3. Ensure the back button on the note takes you back to the note list
4. Test keyboard commands to show/hide note list and move between next/previous notes

### Release
Not updated: Fix a bug that made it impossible to access the note list on small screens

codebykat committed:
most of a fix for this by not overloading CLOSE_NOTE

codebykat committed:
default to true so we show the note list on first load on small screens

codebykat committed:
toggle back into the note list after next/prev

codebykat committed:
don't open note list on next/prev if it's not already open

codebykat committed:
change wording of note list key since it's a toggle now

dmsnell committed:
Remove some unused code

dmsnell committed:
Remove toggling behavior from note list

dmsnell committed:
Separate the action of selecting a note from the action of opening it

dmsnell committed:
Disable note-editing from list view and open note on Enter presses

It was possible to edit the contents of a note in a small-screen view
with the note list open. This was confusing because it wasn't obvious
that the edits were applying.

In this change the editor (with help of some ugly prop-drilling)
disables itself while in a small-screen view while the note list is open.
In order to ease the jump back to the note the Enter key now opens it.

belcherj commented on lib/note-content-editor.tsx:
```diff
@@ -352,6 +357,7 @@ class NoteContentEditor extends Component<Props> {
   };
 
   render() {
+    console.log(this.props.editingEnabled);
```
Remove console.log

belcherj commented on lib/note-detail/index.tsx:
```diff
@@ -249,6 +249,7 @@ export class NoteDetail extends Component<Props> {
     const {
       note,
       fontSize,
+      isSmallScreen,
```
I don't think this is needed due to https://github.com/Automattic/simplenote-electron/pull/2037/files#diff-52e457c4f342fbeecf796739f2492ccaR385

belcherj commented on lib/note-editor/index.tsx:
```diff
@@ -139,14 +140,15 @@ export class NoteEditor extends Component<Props> {
     return (
       <div className="note-editor theme-color-bg theme-color-fg">
         <NoteDetail
+          isSmallScreen={isSmallScreen}
```
I don't think this is needed due to https://github.com/Automattic/simplenote-electron/pull/2037/files#diff-52e457c4f342fbeecf796739f2492ccaR385

codebykat committed:
review cleanup

codebykat commented:
Addressed the review comments and I think this is looking great! Maintaining the currently selected note and using `Enter` to open into note view feels really slick. I think this a lot more usable than it was before honestly.

I have one really minor edge case, which is that on first load in wide screen, a note is seemingly selected (the first note is in the editor window), but if you resize the window to narrow screen, you get the note list. If you instead select a note first and then resize, it sticks with the note view. I think this is probably fine since the user hasn't actively selected a note yet, but it's a little inconsistent.

belcherj reviewed (approved):
Ship It!

dmsnell committed:
Keep highlighted note in view in list view

dmsnell committed:
Stop rendering editor when it's not visible

Previously we were always rendering the editor.
When very large or slow notes were in the list it made for
abrupt pauses while scrolling through the list.

In this change we stop rendering the editor altogether if
only the list view is visible. This has removed all abrupt
pauses when scrolling through the list view on the
narrow-screen mode.

dmsnell committed:
Remove workaround for disabling the editor while hidden
